### ISSUE TITLE
Can't switch camera to CamTwist virtual camera

### ISSUE THREAD

**Issue by WolfgangFahl** [jitsi/jitsi-meet#5335]
## Description
Switching to a CamTwist virtual camera using Version 80.0.3987.149 (Offizieller Build) (64-Bit) on MacOs 10.13.6 does not work. See also #1964

## Current behavior
Settings shows 4 different cameras: 1. HD WebCam C615, 2. FaceTimeHDCamera 3. CamTwist 4. CamTwist 2VUY

Switching to 1 and 2 works. Switching to 3 or 4 fails although preview is shown correctly.

## Expected Behavior
Not only preview works but switching to the camera

## Possible Solution
unclear

## Steps to reproduce
Install CamTwist and configure it e.g. using OBS and following the procedure at https://www.rightpoint.com/thought/2017/12/19/improving-your-video-meetings-on-skype-with-green-screens  
Test in different environments like skype, chome with https://test.webrtc.org/

# Environment details
see description

**Comment by erkswede** [comment]
This affects me too. Can't switch to droidcam virtual camera. The camera shows up and I can see the video stream from within settings, but the it doesn't stick when I click ok.

**Comment by saghul** [comment]
I'm puzzled because I installed CamTwist and cannot reproduce the problem. I can switch just fine.

What browser and version are you using? Can either of you please share your JS console logs?

**Comment by WolfgangFahl** [comment]
@saghul thank you for looking into this. My first test was with firefox 74.0 64 Bit and it simply crashed. On Chrome I get to the point:
![grafik](https://user-images.githubusercontent.com/1336221/77642348-ccfd1080-6f5d-11ea-9ca3-3ace30fff12d.png)

`2020-03-26T11:33:50.393Z [modules/RTC/RTCUtils.js] <t.value>:  Using the new gUM flow
Logger.js:154 2020-03-26T11:33:50.395Z [modules/RTC/RTCUtils.js] <t.<anonymous>>:  Got media constraints:  {video: {…}, audio: false}
Logger.js:154 2020-03-26T11:33:50.419Z [modules/RTC/RTCUtils.js] onUserMediaSuccess
Logger.js:154 2020-03-26T11:33:50.423Z [JitsiConference.js] <oe._doReplaceTrack>:  _doReplaceTrack - no JVB JingleSession
Logger.js:154 2020-03-26T11:33:50.424Z [JitsiConference.js] <oe._doReplaceTrack>:  _doReplaceTrack - no P2P JingleSession
Logger.js:154 2020-03-26T11:33:50.460Z [features/base/tracks] Replace video track - unmuted
Logger.js:154 2020-03-26T11:33:50.499Z [modules/UI/videolayout/LargeVideoManager.js] hover in %s e26d1f93
Logger.js:154 2020-03-26T11:33:50.510Z [conference.js] switched local video device
Logger.js:154 2020-03-26T11:33:50.857Z [modules/UI/videolayout/LargeVideoManager.js] hover in %s e26d1f93`

**Comment by erkswede** [comment]
I'm on chrome 80.0.3987.149. I have the same behavior as @WolfgangFahl 

Can this issue be related to language/locale settings in the browser. I'm located in Sweden.

My log:
```
Logger.js:154 2020-03-26T13:00:01.796Z [modules/RTC/RTCUtils.js] <t.value>:  Using the new gUM flow
Logger.js:154 2020-03-26T13:00:01.796Z [modules/RTC/RTCUtils.js] <t.value>:  Using the new gUM flow
Logger.js:154 2020-03-26T13:00:01.797Z [modules/RTC/RTCUtils.js] <t.<anonymous>>:  Got media constraints:  {video: false, audio: {…}}
Logger.js:154 2020-03-26T13:00:01.798Z [modules/RTC/RTCUtils.js] <t.<anonymous>>:  Got media constraints:  {video: {…}, audio: false}
Logger.js:154 2020-03-26T13:00:01.828Z [modules/RTC/RTCUtils.js] onUserMediaSuccess
Logger.js:154 2020-03-26T13:00:01.879Z [modules/RTC/RTCUtils.js] onUserMediaSuccess
Logger.js:154 2020-03-26T13:00:03.879Z [modules/RTC/RTCUtils.js] <t.value>:  Using the new gUM flow
Logger.js:154 2020-03-26T13:00:03.880Z [modules/RTC/RTCUtils.js] <t.<anonymous>>:  Got media constraints:  {video: {…}, audio: false}
Logger.js:154 2020-03-26T13:00:03.900Z [modules/RTC/RTCUtils.js] onUserMediaSuccess
Logger.js:154 2020-03-26T13:00:04.989Z [modules/RTC/RTCUtils.js] <t.value>:  Using the new gUM flow
Logger.js:154 2020-03-26T13:00:04.990Z [modules/RTC/RTCUtils.js] <t.<anonymous>>:  Got media constraints:  {video: {…}, audio: false}
Logger.js:154 2020-03-26T13:00:05.017Z [modules/RTC/RTCUtils.js] onUserMediaSuccess
Logger.js:154 2020-03-26T13:00:05.018Z [JitsiConference.js] <oe._doReplaceTrack>:  _doReplaceTrack - no JVB JingleSession
Logger.js:154 2020-03-26T13:00:05.019Z [JitsiConference.js] <oe._doReplaceTrack>:  _doReplaceTrack - no P2P JingleSession
```

**Comment by Rik44444** [comment]
I have the same problem, cannot switch cameras. I'm on Windows 10 using Firefox.

**Comment by guevarajoge** [comment]
Same as @WolfgangFahl  n the rest.  I see CamTwist in the dropdown-webcamera-menu but after clicking on it, it will return to my FaceTime default camera.
Although  at some point today it worked fine for a little while ... then i restarted  jitsi and it came back to the faulty behaviour .
Any help ??

**Comment by guevarajoge** [comment]
Good news!  its working on mine :)

Macbook pro 16" -  OS Catalina 10.15.4
Chrome Version 80.0.3987.163
CamTwist Version 3.4.3 (3.4.3)   

**I changed the video size** 
Following the path
CamTwist/Preferences.../General/Video Size/ 1280x720  

By default CamTwist uses 320x240. 
It works fine if the Video Size is set to 1280x720 , I tested this with 2 different macbooks. 

 It make sense that I can see my preview in Jitsi dropdown-webcam-menu because the preview its  "small". 
But to really using the image coming from CamTwist, Jitsi will not accept the tiny 320x240 native resolution of CamTwist. 

I hope this work for you guys !! 




<img width="1792" alt="Screenshot 2020-04-12 at 04 00 43" src="https://user-images.githubusercontent.com/23557957/79058756-3aa18000-7c72-11ea-8204-c36851c8c7d2.png">
